### PR TITLE
fix(RHINENG-4012): Validate custom staleness values are consistent

### PR DESCRIPTION
**Pull request by jpramos123**
# Overview

This PR is being created to address [RHINENG-4012](https://issues.redhat.com/browse/RHINENG-4012).

Validates if staleness input values are consistent.

## PR Checklist

- [ ] Keep PR title short, ideally under 72 characters
- [ ] Descriptive comments provided in complex code blocks
- [ ] Tests: validate optimal/expected output
- [ ] Tests: validate exceptions and failure scenarios
- [ ] Tests: edge cases
- [ ] Recovers or fails gracefully during potential resource outages (e.g. DB, Kafka)
- [ ] Uses [type hinting](https://docs.python.org/3/library/typing.html), if convenient
- [ ] Documentation, if this PR changes the way other services interact with host inventory
- [ ] Links to related PRs

## Secure Coding Practices Documentation Reference

You can find documentation on this checklist [here](https://github.com/RedHatInsights/secure-coding-checklist).

## Secure Coding Checklist

- [ ] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [ ] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [ ] General Coding Practices

### CODE REVIEW

File: app/models.py [hunk 1]
```diff
@@ -869,5 +869,30 @@ class StalenessSchema(MarshmallowSchema):
...
+            raise MarshmallowValidationError(
+                "immutable_time_to_stale must be lower than immutable_time_to_stale_warning"
+            )
+        elif data["immutable_time_to_stale"] >= data["immutable_time_to_delete"]:
+            raise MarshmallowValidationError("immutable_time_to_stale must be lower than immutable_time_to_delete")
+        elif data["immutable_time_to_stale_warning"] >= data["immutable_time_to_delete"]:
+            raise MarshmallowValidationError(
+                "immutable_time_to_stale_warning must be lower than immutable_time_to_delete"
+            )
+
```
Comment (kruai) [inline app/models.py:896]: In order to reduce the repetition here, I suggest something like this:
```suggestion
        staleness_fields = ["time_to_stale", "time_to_stale_warning", "time_to_delete"]
        for host_type in ("conventional", "immutable",):
            for i in range(len(staleness_fields) - 1): # For all but the last field
                for j in range(i+1, len(staleness_fields)): # For all fields after that field
                    if data[(field_1 := f"{host_type}_{staleness_fields[i]}")] >= data[(field_2 := f"{host_type}_{staleness_fields[j]}")]:
                        raise MarshmallowValidationError(
                            f"{field_1} must be lower than {field_2}"
                        )
```

`field_1` and `field_2` can be instantiated on separate lines if L878 is too hard to read.

**Review by kruai — CHANGES_REQUESTED** [review]
Looking good, just a few recs

File: api/staleness.py [hunk 2]
```diff
@@ -30,27 +32,13 @@
...
-
-
 def _validate_input_data(body):
     # Validate account staleness input data
     try:
-        validated_data = StalenessSchema().load(body)
+        identity = get_current_identity()
+        staleness_obj = serialize_staleness_to_dict(get_staleness_obj(identity))
+        data_to_validate = merge_dictionaries(body, staleness_obj)
+        validated_data = StalenessSchema().load(data_to_validate)
```
[the remark this answers is not in the document]
Comment (jpramos123) [reply api/staleness.py:41]: Loved it! Thanks!

File: tests/test_api_staleness_create.py [hunk 3]
```diff
@@ -93,3 +85,11 @@ def test_create_staleness_rbac_denied(subtests, mocker, api_create_staleness, db
...
 
             assert_response_status(response_status, 403)
+
+
+def test_create_improper_staless(api_create_staleness):
+    input_data = {
+        "conventional_time_to_stale_warning": "1",
+    }
+    response_status, response_data = api_create_staleness(input_data)
+    assert_response_status(response_status, 400)
```
[the remark this answers is not in the document]
Comment (jpramos123) [reply tests/test_api_staleness_create.py:95]: Added all scenarios to as parametrize! Thanks

**Review by kruai — APPROVED** [review]
LGTM if you were OK with my `validate_staleness` suggestion! I hope you didn't feel pressured to replace it completely - I wanted to reduce repetition, but we can meet halfway if you think it's a bit hard to read

**Comment by jpramos123** [comment]
> LGTM if you were OK with my `validate_staleness` suggestion! I hope you didn't feel pressured to replace it completely - I wanted to reduce repetition, but we can meet halfway if you think it's a bit hard to read

Naah!! I loved it! I was not able to see this path!! 

Thank you for the suggestion!

**Pull request merged**